### DISCUSSION TITLE
Handling multiline set-env when command errors out

### DISCUSSION THREAD

**Discussion by latacora-tomekr** [community/community#26405] [Actions]
<p>I have the following step defined</p>
<pre><code class="lang-auto">      - name: Multiline Test
        id: multiline
        run: |
          echo 'MULTILINE_OUTPUT&lt;&lt;EOF' &gt;&gt; $GITHUB_ENV
          a-command-that-can-fail &gt;&gt; $GITHUB_ENV
          echo 'EOF' &gt;&gt; $GITHUB_ENV
        continue-on-error: true
</code></pre>
<p>In this case, <code>a-command-that-can-fail</code> may or may not throw an exit code. I’d like the workflow to continue executing and display the command error. However, it appears that if <code>a-command-that-can-fail</code> throws an exit code, the step stops executing the rest of the run. Therefore, <code>echo 'EOF' &gt;&gt; $GITHUB_ENV</code> never runs and I can not access the output.</p>
<p>Is there a recommended way to handle this?</p>

**Comment by Simran-B** [comment]
<p>Depending on the command, there might be a built-in option to disable the non-zero exit code, but as you are interested in this code, one of these options might do what you want:</p>
<aside class="quote" data-post="4" data-topic="126719">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/github-actions-using-grep-when-no-lines-selected/126719/4">GitHub Actions: using grep when no lines selected</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    You can also do one of the following: 

echo "foo" | grep "bar" || true

set +e
echo "foo" | grep "bar"
set -e



- run: |
    echo "foo" | grep "bar"
  continue-on-error: true


Perhaps use a <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#custom-shell" rel="noopener nofollow ugc">Custom shell</a> that doesn’t fail if a command returns a non-zero exit code (-eo pipefail removed)
  </blockquote>
</aside>

**Comment by latacora-tomekr** [comment, marked as answer]
<p>Yea the tricky bit here is that I actually want the step to result in a failed status which I surface in a PR message. I ended up getting passed this by writing the output to a temporary file and reading it the next step.</p>
<pre><code class="lang-auto">     - name: Multiline Test
        id: multiline
        run: |
          a-command-that-can-fail &gt; "$RUNNER_TEMP/output.txt"
        continue-on-error: true
</code></pre>
<p>Then I retrieved the contents in the next step with actions/github-script</p>
<pre><code class="lang-auto">            const tempDirectory = process.env['RUNNER_TEMP']

            const fs = require('fs').promises
            const planContents = await fs.readFile(`${tempDirectory}/output.txt`, {encoding: 'utf8'})
</code></pre>
